test/cets_SUITE.erl

```diff
@@ -2208,10 +2264,340 @@ join_interrupted_when_ping_crashes(Config) ->
+    rpc(Peer2, erlang, disconnect_node, [Node1]),
+    receive_message({nodedown, Node2}),
+    Tab = make_name(Config),
+    {ok, _Pid1} = start(Node1, Tab),
+    {ok, _Pid2} = start(Peer2, Tab),
+    F = fun(State) ->
+        {{ok, [Node1, Node2]}, State}
+    end,
+    DiscoName = disco_name(Config),
+    Disco = start_disco(Node1, #{
+        name => DiscoName, backend_module => cets_discovery_fun, get_nodes_fn => F
+    }),
+    cets_discovery:add_table(Disco, Tab),
+    ok = cets_discovery:wait_for_ready(Disco, 5000),
+    receive_message({nodeup, Node2}),
+    rpc(Peer2, erlang, disconnect_node, [Node1]),
+    receive_message({nodedown, Node2}),
+    receive
+        {log, ?FUNCTION_NAME, #{
+            level := warning,
+            meta := #{pid := Disco},
+            msg := {report, #{what := nodedown, remote_node := Node2} = R}
+        }} = M ->
+            ?assert(is_integer(maps:get(alive_nodes, R)), M),
+            ?assert(is_integer(maps:get(time_since_startup_in_milliseconds, R)), M),
+            ?assert(is_integer(maps:get(connected_millisecond_duration, R)), M)
+    after 5000 ->
+        ct:fail(timeout)
+    end.
+
+disco_node_down_timestamp_is_remembered(Config) ->
+    logger_debug_h:start(#{id => ?FUNCTION_NAME}),
+    Node1 = node(),
+    #{ct2 := Peer2} = proplists:get_value(peers, Config),
+    #{ct2 := Node2} = proplists:get_value(nodes, Config),
+    rpc(Peer2, erlang, disconnect_node, [Node1]),
+    Tab = make_name(Config),
+    {ok, _Pid1} = start(Node1, Tab),
+    {ok, _Pid2} = start(Peer2, Tab),
+    F = fun(State) ->
+        {{ok, [Node1, Node2]}, State}
+    end,
+    DiscoName = disco_name(Config),
+    Disco = start_disco(Node1, #{
+        name => DiscoName, backend_module => cets_discovery_fun, get_nodes_fn => F
+    }),
+    cets_discovery:add_table(Disco, Tab),
+    ok = cets_discovery:wait_for_ready(Disco, 5000),
+    rpc(Peer2, erlang, disconnect_node, [Node1]),
+    %% Check that nodedown is remembered
+    cets_test_wait:wait_until(
+        fun() ->
+            #{nodedown_timestamps := Map} = cets_discovery:system_info(Disco),
+            maps:is_key(Node2, Map)
+        end,
+        true
+    ).
+
+disco_logs_nodeup_after_downtime(Config) ->
+    logger_debug_h:start(#{id => ?FUNCTION_NAME}),
+    Node1 = node(),
+    #{ct2 := Peer2} = proplists:get_value(peers, Config),
+    #{ct2 := Node2} = proplists:get_value(nodes, Config),
+    rpc(Peer2, erlang, disconnect_node, [Node1]),
+    Tab = make_name(Config),
+    {ok, _Pid1} = start(Node1, Tab),
+    {ok, _Pid2} = start(Peer2, Tab),
+    F = fun(State) ->
+        {{ok, [Node1, Node2]}, State}
+    end,
+    DiscoName = disco_name(Config),
+    Disco = start_disco(Node1, #{
+        name => DiscoName, backend_module => cets_discovery_fun, get_nodes_fn => F
+    }),
+    cets_discovery:add_table(Disco, Tab),
+    ok = cets_discovery:wait_for_ready(Disco, 5000),
+    rpc(Peer2, erlang, disconnect_node, [Node1]),
```

Comment:
We check reconnect here.
After disconnect_node on this line, disco should reconnect automatically.
We selectively receive nodeup with downtime_millisecond_duration field.
downtime_millisecond_duration field would only get added after the second nodeup (first one we never extract from the message box).

If we remove this disconnect_node line, the test would fail with the timeout in the receive statement.

I add more comments here into the code though, describing what we test.